docs/proposals/20231122-automate-aks-features.md

```diff
@@ -0,0 +1,311 @@
+as an ASO ManagedCluster:
+
+```go
+type AzureManagedControlPlaneClassSpec struct {
+	...
+	
+	// ManagedClusterSpec defines the spec of the ASO ManagedCluster managed by this AzureManagedControlPlane.
+	ManagedClusterSpec v1api20230201.ManagedCluster_Spec `json:"managedClusterTemplate"`
+}
+```
+
+This method allows CAPZ to leverage schema validation defined by ASO's CRDs upon AzureManagedControlPlane
+creation, but would still lose any further webhook validation done by ASO unless CAPZ can invoke that itself.
+
+It also has the drawback that one version of CAPZ is tied directly to a single AKS API version. The spec could
+potentially contain separate fields for each API version and enforce in webhooks that only one API version is
+being used at a time. Alternatively, users may set fields only present in a newer API version directly on the
+ManagedCluster after creation (if allowed by AKS) because CAPZ will not override user-provided fields for
+which it does not have its own opinion on ASO resources.
+
+#### Decision
+
+TBD
+
+### Security Model
+
+Overall, none of the approaches outlined above change what data is ultimately represented in the API, only the
+higher-level shape of the API. That means there is no further transport or handling of secrets or other
+sensitive information beyond what CAPZ already does.
+
+### Risks and Mitigations
+
+Increasing CAPZ's reliance on ASO and exposing ASO to users at the API level further increases the risk
+[previously discussed](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/ce3c130266b23a8b67aa5ef9a21f257ff9e6d63e/docs/proposals/20230123-azure-service-operator.md?plain=1#L169)
+that since ASO has not yet been proven to be as much of a staple as the other projects that manage
+infrastructure on Azure, ASO's lifespan may be more limited than others. If ASO were to sunset while CAPZ
+still relies on it, CAPZ would have to rework its APIs. This risk is mitigated by the fact that no
+announcements have yet been made regarding ASO's end-of-life and the project continues to be very active.
+
+## Alternatives
+
+The main alternative to leveraging ASO to expose the entire AKS API from CAPZ would be for CAPZ to build its
+own code generation pipeline which takes as input the Azure API specs and ultimately produces definitions for
+AzureManagedControlPlane and AzureManagedMachinePool and mappings to ASO's ManagedCluster and
+ManagedClustersAgentPool. Such a solution would likely end up being too similar to what ASO already provides
+for the effort to be worthwhile.
+
+## Upgrade Strategy
+
+Each of the [options above](#api-design-options) suggest additive, backwards-compatible API changes. Users
+will not need to perform any action after upgrading to the first version of CAPZ that implements this
+proposal.
+
+The new API field allowing the entire ASO resource to be defined in the CAPZ API will overlap with several
```

Comment:
An alternative strategy would be to introduce a new CAPZ API version for ManagedClusters (v1beta2?) and allow users to upgrade by upgrading the API version with conversion webhooks that do the translation from the old fields to the new ASO ones for existing fields

Reply:
Added this to the doc.